realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -413,38 +414,47 @@ private static void initializeRealm(Realm realm) {
         // the Realm is initialized.
         boolean commitChanges = false;
         try {
-            realm.beginTransaction();
+            // We need to start a transaction no matter readOnly mode, because it acts as an interprocess lock.
+            // TODO: For proper inter-process support we also need to move e.g copying the asset file under an
+            // interprocess lock. This lock can obviously not be created by a Realm instance so we probably need
+            // to implement it in Object Store. When this happens, the `beginTransaction(true)` can be removed again.
+            realm.beginTransaction(true);
```

Comment:
But the schema validation procedure might be changed for OS schema integration, not sure what is the best to do right now.